dirs/dirs.go

```diff
@@ -190,6 +198,36 @@ func init() {
 	SetRootDir(root)
 }
 
+// SnapHomeDirs returns a slice of the home directories currently in use.
+func SnapHomeDirs() []string {
+	snapHomeDirsMu.Lock()
+	defer snapHomeDirsMu.Unlock()
+	dirs := make([]string, len(snapHomeDirs))
+	copy(dirs, snapHomeDirs)
```

Comment:
if this respected GlobalRootDir we might need fewer test changes,  when we copy the dirs we could check if they start with GlobalRootDir as path prefix (adding a / to it if needed) and otherwise prepend that to the dir

Reply:
I did this change inside `dirs.SetSnapHomeDirs` to accommodate for the glob expression changes